pkg/internal/server/grpc/common_test.go

```diff
@@ -204,62 +234,176 @@ func (th *testHarness) makeFakeSnapshotClient() *fakesnapshot.Clientset {
+		vsc := th.VolumeSnapshotContent(ga.GetName(), th.DriverName)
 		return true, vsc, nil
 	})
+	fakeSnapshotClient.PrependReactor("get", "volumesnapshotclasses", func(action clientgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
+		ga := action.(clientgotesting.GetAction)
+		if ga.GetName() != th.VolumeSnapshotClassName {
+			return false, nil, nil
+		}
+		vs := th.VSCNoAnn() // Not the default but parameters set
+		vs.Name = ga.GetName()
+		return true, vs, nil
+	})
 	return fakeSnapshotClient
 }
 
-func volumeSnapshot(name, namespace string) *snapshotv1.VolumeSnapshot {
+// FakeCountReactors saves the length of the fake reaction chains.
+// Implicitly called by WithFakeClientAPIs but can be explicitly
+// called if additional "permanent" reactors are added.
+func (th *testHarness) FakeCountReactors() {
+	th.lenFakeKubeClientReactionChain = len(th.FakeKubeClient.ReactionChain)
+	th.lenFakeSnapshotClientReactionChain = len(th.FakeSnapshotClient.ReactionChain)
+	th.lenFakeCBTClientReactionChain = len(th.FakeCBTClient.ReactionChain)
+}
+
+// FakePopPrependedReactors pops reactors from the front of the fake reaction chains.
+// Do not use if AddReactor was used after the call to CountFakeReactors.
+func (th *testHarness) FakePopPrependedReactors() {
+	var numToPop int
+
+	numToPop = len(th.FakeKubeClient.ReactionChain) - th.lenFakeKubeClientReactionChain
+	if numToPop > 0 {
+		th.FakeKubeClient.ReactionChain = th.FakeKubeClient.ReactionChain[numToPop:]
+	}
+
+	numToPop = len(th.FakeSnapshotClient.ReactionChain) - th.lenFakeSnapshotClientReactionChain
+	if numToPop > 0 {
+		th.FakeSnapshotClient.ReactionChain = th.FakeSnapshotClient.ReactionChain[numToPop:]
+	}
+
+	numToPop = len(th.FakeCBTClient.ReactionChain) - th.lenFakeCBTClientReactionChain
+	if numToPop > 0 {
+		th.FakeCBTClient.ReactionChain = th.FakeCBTClient.ReactionChain[numToPop:]
+	}
+}
+
+// ContentNameFromSnapshot returns the content name for a snapshot name.
+func (th *testHarness) ContentNameFromSnapshot(snapshotName string) string {
+	return "snapcontent-" + snapshotName
+}
+
+// SnapshotNameFromContent is the inverse of ContentNameFromSnapshot().
+func (th *testHarness) SnapshotNameFromContent(contentName string) string {
+	return strings.TrimPrefix(contentName, "snapcontent-") // Identity if prefix missing.
+}
+
+// HandleFromSnapshot returns a snapshot identifier based on the snapshot name.
+func (th *testHarness) HandleFromSnapshot(snapshotName string) string {
+	return snapshotName + "-id"
+}
+
+func (th *testHarness) VolumeSnapshot(snapshotName, namespace string) *snapshotv1.VolumeSnapshot {
 	return &snapshotv1.VolumeSnapshot{
 		ObjectMeta: apimetav1.ObjectMeta{
-			Name:      name,
+			Name:      snapshotName,
 			Namespace: namespace,
 		},
 		Spec: snapshotv1.VolumeSnapshotSpec{
-			VolumeSnapshotClassName: stringPtr("csi-snapshot-class"),
+			VolumeSnapshotClassName: stringPtr(th.VolumeSnapshotClassName),
```

Comment:
Needed to set an expected VolumeSnapshotClass name, so these helpers became methods on the test harness.